internal/provider/kubernetes/controller.go

```diff
@@ -249,7 +249,7 @@ func (r *gatewayAPIReconciler) Reconcile(ctx context.Context, request reconcile.
 	}
 
 	// Process the parametersRef of the accepted GatewayClass.
-	if acceptedGC.Spec.ParametersRef != nil && acceptedGC.DeletionTimestamp == nil {
+	if acceptedGC.Spec.ParametersRef != nil {
```

Comment:
I have added the check inside of processParamsRef func to remove finalizer if gatewayClass will be terminated.
https://github.com/envoyproxy/gateway/blob/5797f6731a4846438a1372dfd78ca397ed36f36c/internal/provider/kubernetes/controller.go#L1344

Reply:
the issue is at the end of this `if` where it performs a `r.resources.GatewayAPIResources.Store(acceptedGC.Name, resourceTree)`
we dont want to insert a deleted GC into the resource tree

Reply:
This `if` is used only for the `processParamsRef` function or am I missing something there?
```go
if acceptedGC.Spec.ParametersRef != nil {	
   if err := r.processParamsRef(ctx, acceptedGC, resourceTree); err != nil {
       msg := fmt.Sprintf("%s: %v", status.MsgGatewayClassInvalidParams, err)
       if err := r.gatewayClassUpdater(ctx, acceptedGC, false, string(gwapiv1b1.GatewayClassReasonInvalidParameters), msg); err != nil {
          r.log.Error(err, "unable to update GatewayClass status")
       }
       r.log.Error(err, "failed to process parametersRef for gatewayclass", "name", acceptedGC.Name)
       return reconcile.Result{}, err
   }
}
```
I moved the `acceptedGC.DeletionTimestamp` check inside this function, so if gateway class is being deleted `processParamsRef` will remove the finalizer from EnvoyProxy and return nil (and not add anything to the resourceTree).

Reply:
yup you're right, this flow is fine, the `acceptedGC` at this point is not deleted since it has been checked earlier when computing `acceptedGC`, thanks for the correction